packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol

```diff
@@ -0,0 +1,88 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+/* Testing utilities */
+import "./CommonTest.t.sol";
+import { CrossDomainMessenger } from "../universal/CrossDomainMessenger.sol";
+
+// Tests for obtaining pure gas cost estimates for commonly used functions.
+// The objective with these benchmarks is to strip down the actual test functions
+// so that they are nothing more than the call we want measure the gas cost of.
+// In order to achieve this we make no assertions, and handle everything else in the setUp()
+// function.
+contract GasBenchMark_L1CrossDomainMessenger is Messenger_Initializer {
+    function test_L1MessengerSendMessage_benchmark() external {
+        // The amount of data typically sent during a bridge deposit.
+        bytes
+            memory data = hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+        L1Messenger.sendMessage(bob, data, uint32(100));
+    }
+}
+
+contract GasBenchMark_L1StandardBridge_Deposit is Bridge_Initializer {
+    function setUp() public virtual override {
+        super.setUp();
+        deal(address(L1Token), alice, 100000, true);
+        vm.startPrank(alice, alice);
+    }
+
+    function test_depositETH_benchmark() external {
+        L1Bridge.depositETH{ value: 500 }(50000, hex"");
+    }
+
+    function test_depositERC20_benchmark() external {
+        L1Bridge.depositETH{ value: 500 }(50000, hex"");
+    }
+}
+
+contract GasBenchMark_L1StandardBridge_Finalize is Bridge_Initializer {
+    function setUp() public virtual override {
+        super.setUp();
+        deal(address(L1Token), address(L1Bridge), 100, true);
+        vm.mockCall(
+            address(L1Bridge.messenger()),
+            abi.encodeWithSelector(CrossDomainMessenger.xDomainMessageSender.selector),
+            abi.encode(address(L1Bridge.otherBridge()))
+        );
+        vm.startPrank(address(L1Bridge.messenger()));
+        vm.deal(address(L1Bridge.messenger()), 100);
+    }
+
+    function test_finalizeETHWithdrawal_benchmark() external {
+        // TODO: Make this more accurate. It is underestimating the cost because it pranks
+        // the call coming from the messenger, which bypasses the portal
+        // and oracle.
+        L1Bridge.finalizeETHWithdrawal{ value: 100 }(alice, alice, 100, hex"");
+    }
+}
+
+contract GasBenchMark_OptimismPortal is Portal_Initializer {
+    function setUp() public override {
+        super.setUp();
+    }
+
+    function test_depositTransaction_benchmark() external {
```

Comment:
I recommend running different versions of this test where you are setting the basefee to different values

Reply:
I added the case where the basefee is set to `INITIAL_BASE_FEE`:

https://github.com/ethereum-optimism/optimism/blob/045e66ddd6137f80af2d8c8fca3fca28cccf6a4c/packages/contracts-bedrock/contracts/test/BenchmarkTest.t.sol#L50-L59

Which is much cheaper: 

https://github.com/ethereum-optimism/optimism/blob/045e66ddd6137f80af2d8c8fca3fca28cccf6a4c/packages/contracts-bedrock/.gas-snapshot#L9-L10

Beyond that, I don't have great intuition about what values would be interesting to benchmark. I think this should be good for now though.